examples/screenshot/src/main.rs

```diff
@@ -0,0 +1,81 @@
+#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release
+
+use eframe::{
+    egui::{self, ColorImage},
+    glow::{self, HasContext},
+};
+use itertools::Itertools;
+
+fn main() {
+    let options = eframe::NativeOptions::default();
+    eframe::run_native(
+        "Take screenshots and display with eframe/egui",
+        options,
+        Box::new(|_cc| Box::new(MyApp::default())),
+    );
+}
+
+#[derive(Default)]
+struct MyApp {
+    texture: Option<egui::TextureHandle>,
+    screenshot: Option<ColorImage>,
+}
+
+impl eframe::App for MyApp {
+    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
+        egui::CentralPanel::default().show(ctx, |ui| {
+            if let Some(screenshot) = self.screenshot.take() {
+                self.texture = Some(ui.ctx().load_texture("screenshot", screenshot));
+            }
+
+            if ui
+                .add(egui::Label::new("Screenshot (hover me!):").sense(egui::Sense::hover()))
+                .hovered()
+            {
+                ctx.set_visuals(egui::Visuals::dark());
+            } else {
+                ctx.set_visuals(egui::Visuals::light());
+            };
+            if let Some(texture) = self.texture.as_ref() {
+                ui.image(texture, texture.size_vec2());
+            } else {
+                ui.spinner();
+            }
+
+            ctx.request_repaint();
+        });
+    }
+
+    #[allow(unsafe_code)]
+    fn post_rendering(&mut self, screen_size_px: [u32; 2], frame: &eframe::Frame) {
+        if let Some(gl) = frame.gl() {
```

Comment:
Taking a screenshot each frame is very slow, especialy in debug builds. I think it would be better to only do it when clicking a button.

Reply:
I like the continuous mode, so I made it toggable :)